### ISSUE TITLE
[Core] Admin | Sync Template v3 (Flows) - Did not check for owner existing when setting Derived Owner

### ISSUE THREAD

**Issue by albertocaa1317** [microsoft/coe-starter-kit#603]
After to install the lastest version of CoE Starter kit (June'21) I have everyday this error "Action 'Get_Flow_from_CoE' failed" in the cloud flow "Admin | Sync Template v3 (Flows)"

![image](https://user-images.githubusercontent.com/57954993/121479869-c32cc500-c9ca-11eb-8afc-e20e9e49633f.png)

Raw output:
{
    "statusCode": 404,
    "headers": {
        "x-ms-service-request-id": "e0c626a6-4f7a-4abe-9b3c-5c2c4ced5178,dc5dd613-5e5c-4b3f-ad36-f34389c19216",
        "Cache-Control": "no-cache",
        "Set-Cookie": "ARRAffinity=07c0cf7f0e9ec1005b77f7396dd8339d8359dd81a4b147b8f5e4333471ab90bc; domain=org58a90771.crm4.dynamics.com; path=/; secure; HttpOnly,ReqClientId=624f90da-2444-485a-aea9-e72631b3ba51; expires=Wed, 10-Jun-2071 03:45:20 GMT; path=/; secure; HttpOnly; samesite=none,ARRAffinity=07c0cf7f0e9ec1005b77f7396dd8339d8359dd81a4b147b8f5e4333471ab90bc; domain=org58a90771.crm4.dynamics.com; path=/; secure; HttpOnly",
        "Strict-Transport-Security": "max-age=31536000; includeSubDomains",
        "REQ_ID": "dc5dd613-5e5c-4b3f-ad36-f34389c19216,dc5dd613-5e5c-4b3f-ad36-f34389c19216",
        "AuthActivityId": "590105d0-2c31-442e-b614-d365a484bc3f",
        "x-ms-ratelimit-time-remaining-xrm-requests": "1,199.51",
        "x-ms-ratelimit-burst-remaining-xrm-requests": "5978",
        "OData-Version": "4.0",
        "X-Source": "4723512891968225417201391762391911822115960158101941701795962182238412124512049182,3618135891911192337117821252496424283181541802051271296923623811510815176202758377",
        "Public": "OPTIONS,GET,HEAD,POST",
        "Timing-Allow-Origin": "*",
        "Date": "Thu, 10 Jun 2021 03:45:19 GMT",
        "Content-Length": "383",
        "Allow": "OPTIONS,GET,HEAD,POST",
        "Content-Type": "application/json; odata.metadata=full",
        "Expires": "-1"
    },
    "body": {
        "error": {
            "code": "0x80040217",
            "message": "admin_Maker With Id = 507292f1-6164-433e-b6ac-ec7b17e28fa4 Does Not Exist",
            "@Microsoft.PowerApps.CDS.HelpLink": "http://go.microsoft.com/fwlink/?LinkID=398563&error=Microsoft.Crm.CrmException%3a80040217&client=platform",
            "@Microsoft.PowerApps.CDS.InnerError.Message": "admin_Maker With Id = 507292f1-6164-433e-b6ac-ec7b17e28fa4 Does Not Exist"
        }
    }
}

**Comment by JeneferM-MSFT** [comment]
For future assistance, please show a broader view in your image of where the failure occures, this makes us finding the point of failure easier as we dont remember which action names are where. Thank you!

**Comment by JeneferM-MSFT** [comment]
Can you please help me understand this failure? 
Can you please open the Set Flow Permissions app in the Core solution and browse to this flow and see what shows up for its ownership? 

HOW TO:
You can get the environment name from the trigger of the run
![image](https://user-images.githubusercontent.com/62659043/121533182-45ae8800-c9bd-11eb-8c21-53379d91a21a.png)
And the flow name from Get Flow as Admin
![image](https://user-images.githubusercontent.com/62659043/121533391-77bfea00-c9bd-11eb-960e-523096dcdcc8.png)
![image](https://user-images.githubusercontent.com/62659043/121533526-9aea9980-c9bd-11eb-8376-de34406f3e3a.png)
Then open the Set Flow Permissions app, find that flow, and click its name to get to the permissions screen
![image](https://user-images.githubusercontent.com/62659043/121533707-c5d4ed80-c9bd-11eb-88c5-e15e4fddac38.png)
![image](https://user-images.githubusercontent.com/62659043/121533760-d2f1dc80-c9bd-11eb-9208-106f371369ba.png)

**Comment by albertocaa1317** [comment]
Sorry Jenefer, in the next comunication I'll share a more complete screenshot.

I followed your steps and this is the two flows with the error:
![image](https://user-images.githubusercontent.com/57954993/121538330-ff7d1700-ca04-11eb-8b93-999c60dda58f.png)

![image](https://user-images.githubusercontent.com/57954993/121538672-410dc200-ca05-11eb-8c9a-922022ea6ce7.png)

What is the problem?
It seems to be that the owner doesn't exists in the organization, but there are other flows on this situacion and they doesn't fails.
For example:
![image](https://user-images.githubusercontent.com/57954993/121539213-c1ccbe00-ca05-11eb-8b90-499c6a9effaa.png)

**Comment by JeneferM-MSFT** [comment]
We added new logic to deduce what we call the "Derived Owner" for flows this month. This to get around the fact that the product will not let you re-assign ownership, so we simulate it in the inventory. 

This is failing on the step where we mark up Derived Owner to be the creator if its blank. 
To get around this, I need to first check if the creator is still with the organization. 
We do this everywhere else that we use the field, and I just forgot to do it here. 

Thanks for helping make CoE great! I will get this fixed for the July release.

**Comment by JeneferM-MSFT** [comment]
You have installed Core in a production environment yes? (as apposed to Dataverse for Teams). 
I'm creating a PR and I'm hoping you will install to make sure my fix works for you.

**Comment by JeneferM-MSFT** [comment]
Here is the PR which will work if you are not installed in DfT. 
Can you please install it following the instructions for [installing updates](https://docs.microsoft.com/power-platform/guidance/coe/setup#installing-updates)? Note that if you do not remove the unmanaged layers as described there you will not receive updates from us.  

[CenterofExcellenceCoreComponents_2_1_72_managed.zip](https://github.com/microsoft/coe-starter-kit/files/6633405/CenterofExcellenceCoreComponents_2_1_72_managed.zip)

Then let me know if the issue is resolved, if so this will ship in July (and you can be unblocked in the meantime :))

**Comment by albertocaa1317** [comment]
Hi Jenefer!
Yes, I have the CoE in a Production Enviroment.

PR installed. I will give you feedback asap.

Thanks for your nice job!

**Comment by albertocaa1317** [comment]
The lastest execution of this flow after to apply the v2.1.72 was OK.
Thanks!

**Comment by JeneferM-MSFT** [comment]
Wonderful. Thank you for testing! I will get you a new PR once Manuela finishes the work for the DLP issue so you arent in a broken state with the PR, and then please be sure to install July to get you back inline with everyone else bits wise. 
Thanks for using CoE!

**Comment by albertocaa1317** [comment]
Hi @JeneferM-MSFT,
Can I do something to repair the app DLP Editor V2 thar is broken after to install the PR v2.1.72?
Thanks.

**Comment by JeneferM-MSFT** [comment]
Here is a new PR that has both your issues and the issues from the DLP editor fixed. :) 
[CenterofExcellenceCoreComponents_2_1_75_managed.zip](https://github.com/microsoft/coe-starter-kit/files/6677199/CenterofExcellenceCoreComponents_2_1_75_managed.zip)

**Comment by albertocaa1317** [comment]
Hi @JeneferM-MSFT,
The DLP editor issue is not fixed :(
![image](https://user-images.githubusercontent.com/57954993/122636696-d8040980-d0ea-11eb-8beb-f69509192181.png)

I attach de Monitor log
[PowerAppsTraceEvents (1).txt](https://github.com/microsoft/coe-starter-kit/files/6680312/PowerAppsTraceEvents.1.txt)

**Comment by JeneferM-MSFT** [comment]
Thanks Alberto. It turns out there is an issue when you use the app in the player due to some settings that got flipped. We were informed yesterday by another person with the PR. You should be able to use the app in edit mode while you await July. Please let us know

**Comment by albertocaa1317** [comment]
In edit mode the app doesn't works also but don't worries, I'll wait to the July release. Thanks.

**Comment by JeneferM-MSFT** [comment]
same fix applied to Gov solution's Start Approval for Flows

**Comment by JeneferM-MSFT** [comment]
Fixed in July release. 
Please install [the July release](https://aka.ms/CoEStarterKitDownload) following the instructions for [installing updates](https://docs.microsoft.com/power-platform/guidance/coe/setup#installing-updates). Note that if you do not remove the unmanaged layers as described there you will not receive updates from us.
